acme/acme/client.py

```diff
@@ -66,15 +66,13 @@ def __init__(self, directory, key, alg=jose.RS256, verify_ssl=True,
     @classmethod
     def _regr_from_response(cls, response, uri=None, new_authzr_uri=None,
                             terms_of_service=None):
-        terms_of_service = (
-            response.links['terms-of-service']['url']
-            if 'terms-of-service' in response.links else terms_of_service)
+        if 'terms-of-service' in response.links:
+            terms_of_service = response.links['terms-of-service']['url']
+        if 'next' in response.links:
+            new_authzr_uri = response.links['next']['url']
 
         if new_authzr_uri is None:
-            try:
-                new_authzr_uri = response.links['next']['url']
-            except KeyError:
-                raise errors.ClientError('"next" link missing')
+            raise errors.ClientError(response, 'missing "new_authrz_uri"')
```

Comment:
I'm not saying `response` argument is wrong in general. I'm saying it's not in the scope for this PR :) Let's discuss it elsewhere and fix all occurrences at once instead of deviating in one case.

Other error messages in this file (and entire library) reference primitives from the spec rather than API itself. In fact there is another instance of `"next" link is missing` in this file. Again, let's keep things consitent.